src/Stratis.Bitcoin.Features.Consensus/ProvenBlockHeaders/RewindDataIndexStore.cs

```diff
@@ -32,57 +38,80 @@ public class RewindDataIndexStore : IRewindDataIndexStore
         /// </summary>
         private readonly BackendPerformanceCounter performanceCounter;
 
-        public RewindDataIndexStore(
-            IDateTimeProvider dateTimeProvider,
-            IRewindDataIndexRepository rewindDataIndexRepository)
+        public RewindDataIndexStore(IDateTimeProvider dateTimeProvider, ICoinView coinView, IChainedHeaderTree chainedHeaderTree)
         {
             Guard.NotNull(dateTimeProvider, nameof(dateTimeProvider));
-            Guard.NotNull(rewindDataIndexRepository, nameof(rewindDataIndexRepository));
+            Guard.NotNull(coinView, nameof(coinView));
+            Guard.NotNull(chainedHeaderTree, nameof(chainedHeaderTree));
 
-            this.items = new ConcurrentDictionary<string, (int, bool)>();
-            this.rewindDataIndexRepository = rewindDataIndexRepository;
+            this.items = new ConcurrentDictionary<string, int>();
+            this.coinView = coinView;
+            this.chainedHeaderTree = chainedHeaderTree;
 
             this.performanceCounter = new BackendPerformanceCounter(dateTimeProvider);
         }
 
+        public async Task InitializeAsync()
+        {
+            ChainedHeader tip = this.chainedHeaderTree.GetBestPeerTip();
```

Comment:
we don't have peers connected when you initialize, did we?
shouldn't we get blockstore tip as reference?

Reply:
This might have been meant as `GetConsensusTip`. But I think the block store tip is preferable, as on startup consensus will be unwound to the block store height anyway if it consensus was higher.

Reply:
yes actually I see GetConsensusTip is private while GetBestPeerTip look every peer except for current peer (our fullnode)

Anyway I saw that when we call InitializeAsync we have already the chain tip, so we can just pass that

Reply:
yeah this should be `GetConsensusTip`